apps/framework-cli/src/framework/core/plan.rs

```diff
@@ -90,18 +91,36 @@ pub async fn reconcile_with_reality<T: OlapOperations>(
 ) -> Result<InfrastructureMap, PlanningError> {
     info!("Reconciling infrastructure map with actual database state");
 
+    // Clone the map so we can modify it
+    let mut reconciled_map = current_infra_map.clone();
+    reconciled_map.default_database = project.clickhouse_config.db_name.clone();
+
+    if current_infra_map
+        .tables
+        .iter()
+        .any(|(id, t)| id != &t.id(&project.clickhouse_config.db_name))
+    {
+        // fix up IDs where in the old version it does not contain the DB name
+        let existing_tables = mem::take(&mut reconciled_map.tables);
```

Comment:
Why do you need this?

Reply:
do you mean `mem::take` or the logic?

Reply:
mem::take

Reply:
to move contents of the table out of `reconciled_map.tables`